tests/unit/test_runner.py

```diff
@@ -0,0 +1,201 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Test cases of Runner class."""
+
+import secrets
+import unittest
+from pathlib import Path
+from unittest.mock import MagicMock
+
+import pytest
+
+from errors import RunnerCreateError
+from runner import Runner, RunnerClients, RunnerConfig, RunnerStatus
+from runner_type import GitHubOrg, GitHubRepo, VirtualMachineResources
+from tests.unit.mock import (
+    MockLxdClient,
+    MockRepoPolicyComplianceClient,
+    mock_lxd_error_func,
+    mock_runner_error_func,
+)
+
+
+@pytest.fixture(scope="module", name="vm_resources")
+def vm_resources_fixture():
+    return VirtualMachineResources(2, "7Gib", "10Gib")
+
+
+@pytest.fixture(scope="function", name="token")
+def token_fixture():
+    return secrets.token_hex()
+
+
+@pytest.fixture(scope="function", name="binary_path")
+def binary_path_fixture(tmp_path: Path):
+    return tmp_path / "test_binary"
+
+
+@pytest.fixture(scope="module", name="instance", params=["Running", "Stopped", None])
+def instance_fixture(request):
+    if request.param[0] is None:
+        return None
+
+    attrs = {"status": request.param[0], "execute.return_value": (0, "", "")}
```

Comment:
Same comment here about the possibility of exit_code being a parameter to test execution error case.

Reply:
Minor comments